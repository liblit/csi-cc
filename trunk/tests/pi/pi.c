// NOTE: this example adopted from
// http://www.dartmouth.edu/~rc/classes/soft_dev/C_simple_ex.html

#include <stdio.h>

#include "rand.h"
#define RAND_MAX 10000

int main()
{
  int niter=0;
  double x,y;
  int i,count=0; /* # of points in the 1st quadrant of unit circle */
  double z;
  double pi;
  
  printf("Enter the number of iterations used to estimate pi: ");
  scanf("%d",&niter);
  
  count=0;
  for (i = 0; i < niter; i++){
    x = (double)rand(RAND_MAX)/RAND_MAX;
    y = (double)rand(RAND_MAX)/RAND_MAX;
    z = x*x+y*y;
    if(z <= 1)
      count++;
  }
  pi=(double)count/niter*4;
  printf("# of trials= %d , estimate of pi is %g \n",niter,pi);
}
