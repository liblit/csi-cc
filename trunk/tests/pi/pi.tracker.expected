#
rand
1|EXIT
0|ENTRY|1|1|1|1|3|3|3|3|3|3|-1|3
$
0->1|0$0
#
main
3|EXIT
2|ENTRY|11|11|11|11|11|12|12|13|13|13|14|15|17|18|20|21
4|21|21|21|21
5|22|22|22|22|23|23|23|23|24|24|24|24|24|24|24|24|25|25|25
6|-1|28|28|28|28|28|28|28|29|29|29|30|30
7|26|26|26
10|NULL
8|NULL
9|21|21|21|-1
$
2->4|0$0
4->5|0$0
4->6|2$2
5->7|0$0
5->10|1$1
6->3|0$0
7->8|0$0
10->8|0$0
8->9|0$0
9~>4|3$3
